pkg/standalone/standalone.go

```diff
@@ -225,8 +226,7 @@ func makePodSourceConfig(kc *KubeletConfig) *config.PodConfig {
 		glog.Infof("Adding manifest url: %v", kc.ManifestURL)
 		config.NewSourceURL(kc.ManifestURL, kc.HttpCheckFrequency, cfg.Channel(kubelet.HTTPSource))
 	}
-
-	if kc.EtcdClient != nil {
+	if !reflect.ValueOf(kc.EtcdClient).IsNil() {
```

Comment:
That's one solution, yes. Or you could just check that you didn't get a nil *tools.EtcdClient before storing it in kc.EtcdClient.

reflect here works, but we do NOT want to get into the habit of making people open up interfaces with reflect to do a simple nil check. It's hard enough to remember to check for nils as it is. Allowing for the possibility of needing this reflect line is a way to get subtle, rare bugs.

The C++ analog is passing around a void** when a void* would work fine. Language purists might argue that you shouldn't be able to have a nil at all, but we don't have to go that far to agree that checking for two nils is worse than checking for one nil.

Please also note that @dchen1107's fix here only checks the inner value and will crash if it ever gets a true nil interface value. See: http://play.golang.org/p/gJIcCiqf0r